web/src/lib/types.ts

```diff
@@ -67,19 +67,30 @@ export interface IPendingProofResponse {
   status: "pending" | "mined" | "new";
 }
 
-export type ActionKioskType = Pick<
+export type ActionKioskQueryType = Pick<
   ActionModel,
   "id" | "name" | "description" | "action" | "external_nullifier" | "__typename"
 > & {
-  app: Pick<
-    AppModel,
-    | "id"
-    | "name"
-    | "verified_app_logo"
-    | "is_staging"
-    | "is_verified"
-    | "__typename"
-  >;
+  app: Pick<AppModel, "id" | "is_staging" | "__typename"> & {
+    app_metadata: Array<
+      Pick<AppMetadataModel, "name" | "logo_img_url" | "status">
+    >;
+    verified_app_metadata: Array<
+      Pick<AppMetadataModel, "name" | "logo_img_url" | "status">
+    >;
+  };
+};
+
+export type ActionKioskType = Omit<ActionKioskQueryType, "app"> & {
+  app: Omit<
+    ActionKioskQueryType["app"],
+    "app_metadata" | "verified_app_metadata"
+  > & {
+    app_metadata: Pick<
+      AppMetadataModel,
+      "name" | "logo_img_url" | "status"
```

Comment:
logo_image_url may not be present (in case of unverified apps)

Reply:
yeah but will be empty string so this won't break anything and is good for consistency